### PR TITLE
Allow for the qc error state

### PR DESCRIPTION
**Pull request by ElvisASC4**

### CODE REVIEW

File: ui/src/components/RecordButton.js [hunk 1]
```diff
@@ -146,6 +149,21 @@ export class RecordButton extends LitElement {
         this.dispatchEvent(event);
     }
 
+    /**
+     * Emits an event that causes the application to render thee QCError
```
Comment (ashleygranitto) [inline ui/src/components/RecordButton.js:153]: the instead of thee

File: ui/src/StateContainer.js [hunk 2]
```diff
@@ -168,6 +169,15 @@ export class StateContainer extends LitElement {
         this.context = e.detail.context;
     }
 
+    /**
+     * Updates the qcError property with the most up-to-date
+     * data from the vox-record-button component.
+     */
+    handleQCError(e) {
```
Comment (ashleygranitto) [inline ui/src/StateContainer.js:176]: nit: add @param documentation for e

File: ui/src/components/RecordButton.js [hunk 3]
```diff
@@ -46,6 +47,7 @@ export class RecordButton extends LitElement {
 
     updated() {
         this.handleWaveCanvas();
+        this.handleQCError();
```
Comment (ashleygranitto) [inline ui/src/components/RecordButton.js:50]: I feel like this function will be called very often unnecessarily--is there a certain property you could check if changed before calling it?

File: ui/src/utils/QualityControl.js [hunk 4]
```diff
@@ -47,7 +47,7 @@ export class QualityControl {
 
         if (this.audioBuffer.duration < 2.0) {
             qualityResult.success = false;
-            qualityResult.errorMessage += 'Audio recording failed: recording was too short. Try again';
+            qualityResult.errorMessage += 'Recording is too short. Try again!';
```
Comment (ashleygranitto) [inline ui/src/utils/QualityControl.js:50]: nit: should it be = instead of +=?

File: ui/src/StateContainer.js (hunk 2 above)
Comment (mmallett) [inline ui/src/StateContainer.js:176]: nit name the function `handleQcError`

File: ui/src/components/RecordButton.js (hunk 3 above)
Comment (mmallett) [reply ui/src/components/RecordButton.js:50]:
```javascript
updated(changedProperties) {
  if (changedProperties.has('qcError') {
    this.handleQcError();
  }
}
```

https://lit-element.polymer-project.org/guide/lifecycle#updated

File: ui/src/components/RecordButton.js [hunk 5]
```diff
@@ -146,6 +149,21 @@ export class RecordButton extends LitElement {
         this.dispatchEvent(event);
     }
 
+    /**
+     * Emits an event that causes the application to render thee QCError
+     */
+    handleQCError() {
+        const event = new CustomEvent('update-QC-Error', {
```
Comment (mmallett) [inline ui/src/components/RecordButton.js:156]: event names should be lower case, kebab case

`update-qc-error`

File: ui/src/utils/QualityControl.js (hunk 4 above)
Comment (ElvisASC4) [reply ui/src/utils/QualityControl.js:50]: Either one works but I'll change it

**Pull request merged**